### PR TITLE
Music Player

### PR DESCRIPTION
**Pull request by ZappoMan**
This is an example of a music player entity. This includes a "spawner" script that can be attached to any entity, and when that entity is clicked a new music player will be spawned.

The music player itself works by setting a URL in the userData to point to a json file that matches the google spreadsheet formatted in json. Your spreadsheet should have 5 columns in this order: Song Title, Artist, Album Title, URL to Song WAV file, URL to album art.

The music player works as follows:
- When the music player is "equipped" it will attach headphones to your avatars head.
- After being equipped, you can use your hand controllers joystick to play/pause, and skip songs. Press down on the joystick to play or pause, and press left or right on the joystick to skip to next or previous track.

Notes on making playlists in Google Spreadsheet:
- Here's an example spreadsheet with the correct format: https://docs.google.com/spreadsheets/d/15k8IgYErDHuy7BW9Sx_uLqjrBVZGt5an6kCwRZLw9UE/pubhtml
- for this spreadsheet you can access a JSON version of that sheet with the URL format: https://spreadsheets.google.com/feeds/cells/15k8IgYErDHuy7BW9Sx_uLqjrBVZGt5an6kCwRZLw9UE/od6/public/basic?alt=json
- change the userData of the spawned musicPlayer to the following:
```
{
    "musicPlayer": {
        "playlistURL": "https://spreadsheets.google.com/feeds/cells/15k8IgYErDHuy7BW9Sx_uLqjrBVZGt5an6kCwRZLw9UE/od6/public/basic?alt=json",
        "songVolume": 0.1
    }
}
```

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2f92a4cccdfde90b9ead3c86c0c77df9e864a645 available at: https://gist.github.com/c636abc4ed27bbd04bd8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7254.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7254.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 168fd2d72f661a70b70dbb701abc0e90072980ad available at: https://gist.github.com/c636abc4ed27bbd04bd8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7254.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7254.dmg

File: examples/toybox/musicPlayer/musicPlayer.js [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+            this.entityID = entityID;
+
+            // Get the entities userData property, to see if someone has overridden any of our default settings
+            var userDataText = Entities.getEntityProperties(entityID, ["userData"]).userData;
+            //print(userDataText);
+            var userData = {};
+            if (userDataText !== "") {
+                //print("calling JSON.parse");
+                userData = JSON.parse(userDataText);
+                //print("userData:" + JSON.stringify(userData));
```
Comment (imgntn) [inline examples/toybox/musicPlayer/musicPlayer.js]: have you run into any reliability issues accessing userData during preload?  i've been doing an initializer pattern to wait until userData is there for sure before trying to parse it.  curious whether you've run into that

**Comment by cozza13** [comment]
this is very nice. I have tested it

**Pull request merged**